### PR TITLE
Improve board

### PR DESCRIPTION
**Pull request by wildlifeniki**
## Description
- Limited user input for tiles. Users can now only enter letters, and any lowercase letters are auto-capitalized.
- Host sends out updates when a change was made and typing paused for longer than 2 seconds (This is an additional condition, so updates are still sent on pressing return or selecting a new cell)
- Games screen auto-refreshes with a fading animation. Switching tabs, closing a game, starting a new game, and accepting/denying an invite all initiate this refresh
- Game board is now initialized on creation (pressing start game generates the board) instead of first open. Decreased load time for opening, but added load time after start game is pressed. 
- Used a third-party library to improve the swipe action on invites. Now users can tap the check to accept (like before), but to delete, users now need to swipe to the right to reveal a delete button, which can then be tapped to remove the invite.
- Cleaned up algorithm code by adding a helper method and removed repeated methods (tile-related methods like getTile, setTile, etc.) between the newGame viewController and the inGame view Controller using an interface

## Milestones
Changes made to gameplay in board, board instantiation, and auto refresh

## Test Plan

https://user-images.githubusercontent.com/22784306/183822357-26cdf401-8593-4280-8515-a850c47828bf.mov

denying invite:

https://user-images.githubusercontent.com/22784306/183822411-4ae1d07f-158c-49a6-85a5-219ffe8a7802.mov

accepting invite:

https://user-images.githubusercontent.com/22784306/184014367-e80505d8-e064-4efa-807c-d1ad06a5c35e.mov

### CODE REVIEW

**Comment by tbram1** [comment]
I like the new library for swiping to delete, but what about accept? I left a comment on that commit already that its confusing to see a check mark icon that isnt a button there.

File: meta_capstone/ViewControllers/UIViewController+TileMethods.h [hunk 1]
```diff
@@ -15,7 +15,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, strong) NSMutableArray *tilesArray;
 @property (strong, nonatomic) PFObject *emptyTile;
 @property (strong, nonatomic) NSMutableArray *usableWords;
-@property (nonatomic, strong) NSDictionary *wordCluePairs;
+@property (nonatomic, strong) NSMutableDictionary *wordCluePairs;
```
Comment (tbram1) [inline meta_capstone/ViewControllers/UIViewController+TileMethods.h:18]: why did you change it to mutable?

Comment (wildlifeniki) [reply meta_capstone/ViewControllers/UIViewController+TileMethods.h:18]: I add the word clue pairs one by one, so the dictionary changes and needs to be mutable
